File: 2026POTD/JANUARY/09-01-2026/865.dart
class Solution {
  TreeNode? subtreeWithAllDeepest(TreeNode? root) {
    List dfs(TreeNode? node) {
      if (node == null) return [0, null];
      var l = dfs(node.left);
      var r = dfs(node.right);
      if (l[0] > r[0]) return [l[0] + 1, l[1]];
      if (r[0] > l[0]) return [r[0] + 1, r[1]];
      return [l[0] + 1, node];
    }

    return dfs(root)[1];
  }
}

File: 2026POTD/JANUARY/09-01-2026/865.swift
class Solution {
    func subtreeWithAllDeepest(_ root: TreeNode?) -> TreeNode? {
        func dfs(_ node: TreeNode?) -> (Int, TreeNode?) {
            guard let node = node else { return (0, nil) }
            let (ld, ln) = dfs(node.left)
            let (rd, rn) = dfs(node.right)
            if ld > rd { return (ld + 1, ln) }
            if rd > ld { return (rd + 1, rn) }
            return (ld + 1, node)
        }
        return dfs(root).1
    }
}